src/org/thoughtcrime/securesms/ReactionsHandler.java

```diff
@@ -0,0 +1,151 @@
+package org.thoughtcrime.securesms;
+
+
+import android.content.Context;
+import android.database.Cursor;
+import android.os.Parcel;
+
+import org.thoughtcrime.securesms.database.Address;
+import org.thoughtcrime.securesms.database.DatabaseFactory;
+import org.thoughtcrime.securesms.database.MessageReactionDatabase;
+import org.thoughtcrime.securesms.database.model.MessageRecord;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ReactionsHandler {
+    private MessageReactionDatabase reactionsDb;
+
+    public ReactionsHandler(Context context) {
+        this.reactionsDb = DatabaseFactory.getMessageReactionDatabase(context);
+    }
+
+    /**
+     * NOT TEST
+     * This method only used by the conversation fragment since it will automatically
+     * compute the reactorID's to the user itself, Please uses another method if the reaction made
+     * by the recipient. such method can be accessed directly from the MessageReactionDB class
+     * The reaction time will be set to the current system time in milliseconds
+     * @param record
+     * @param reaction
+     */
+    public void reactToMessage(MessageRecord record, String reaction, Long time) {
```

Comment:
if these two methods have super strict use cases, it might be good to make the names more scary.

`reactToMessageFromConversationFragmentAsSender`